### ISSUE TITLE
Scrolling on android appears to get stuck between views unless moved slowly

### ISSUE THREAD

**Issue by droplessjake** [archriss/react-native-snap-carousel#889]
### Is this a bug report, a feature request, or a question?

Bug Report

### Have you followed the required steps before opening a bug report?

(Check the step you've followed - put an `x` character between the square brackets (`[]`).)

- [x] I have read [the guidelines regarding bug report](https://github.com/meliorence/react-native-snap-carousel/blob/master/CONTRIBUTING.md).
- [x] I have reviewed [the documentation](https://github.com/meliorence/react-native-snap-carousel/blob/master/README.md) in its entirety, including the dedicated documentations :books:.
- [x] I have searched for [existing issues](https://github.com/meliorence/react-native-snap-carousel/issues) and made sure that the problem hasn't already been reported.
- [x] I am using [the latest plugin version](https://github.com/meliorence/react-native-snap-carousel/releases).
- [x] I am following [the issue template](https://raw.githubusercontent.com/meliorence/react-native-snap-carousel/master/ISSUE_TEMPLATE.md) closely in order to produce a useful bug report.

### Have you made sure that it wasn't a [React Native bug](https://github.com/meliorence/react-native-snap-carousel/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22react-native+bug%22+)?

Yes

### Is the bug specific to iOS or Android? Or can it be reproduced on both platforms?

Android 

### Is the bug reproductible in a production environment (not a debug one)?

I have confirmed this by building the android app with NODE_ENV=production in my env file, yes.

### Environment

  Environment:
    React: 17.0.1
    React native: 0.64.2
    react-native-snap-carousel: 3.9.1

  Target Platform:
    Android (12.5.3)

### Expected Behavior

Swiping between views on Android should snap to the next view when released.

### Actual Behavior

If I swipe with some (pretty normal) momentum, the carousel will get stuck between 2 pages, and not call the onSnapToItem method.
If I swipe slowly and release about halfway, it perfectly snaps perfectly to the next page.
I am aware of the performance issues on android and have tried many props/settings and fixes found in various places on the internet and in the issues here, apologies if I've missed something very common.

### Reproducible Demo

(Paste the link to a [Snack example](https://snack.expo.io/) in which the issue can be reproduced. Please follow [the guidelines](https://stackoverflow.com/help/mcve) for providing a Minimal, Complete, and Verifiable example.)

### Steps to Reproduce

These are the current settings after all my efforts.
```
          <Carousel
            scrollEnabled={calendarState === CalendarState.Open}
            firstItem={prevMonths}
            data={monthData}
            sliderWidth={calendarWidth}
            itemWidth={calendarWidth}
            inactiveSlideOpacity={0.5}
            inactiveSlideScale={0.8}
            inactiveSlideShift={0}
            removeClippedSubviews={false}
            lockScrollWhileSnapping={isAndroid}
            enableMomentum={false}
            disableIntervalMomentum
            enableSnap
            onSnapToItem={onMonthChanged}
            renderItem={renderItem}
          />
```

https://user-images.githubusercontent.com/83917938/140102710-343fc1d4-5cc4-450a-8041-dd1ebec5324f.mp4

I am using the carousel to create a calendar, each page is a "month" containing rows of weeks, containing 7 days, there is a lot of elements here when you consider we have potentially 42 "days" per month/page and in my case, up to 7 months at a time....

**Comment by hammlee** [comment]
exactly same problem

**Comment by droplessjordan** [comment]
Same issue here!

**Comment by droplessjake** [comment]
Found this library [react-native-anchor-carousel](https://github.com/lehoangnam97/react-native-anchor-carousel) which is based off/inspired by this one, but seems to be more refined and lightweight and doesnt have these issues. (and is being actively supported).
Would recommend it instead to anyone who finds this thread and needs a simple solution. 👍

**Comment by hammlee** [comment]
I think it's an issue caused by the RN version. I downgraded the version from 0.64.1 to 0.63.4 and it didn't happen anymore.

**Comment by Isaccobosio** [comment]
Hi, as it said in [their guide](https://github.com/meliorence/react-native-snap-carousel/blob/master/doc/TIPS_AND_TRICKS.md#momentum), you should put those props

```
...
enableMomentum={false}
decelerationRate={0.9}
...
```

Hope it helps. This is how I fix this problem

**Comment by Romick2005** [comment]
@droplessjake Can you please try with the latest not published v4 betta version of this library? Just put 
`"react-native-snap-carousel": "git+https://github.com/meliorence/react-native-snap-carousel.git#59fa8950eeb8446124454ab3e2711038066115bb"`
in your package.json and install node dependencies.

**Comment by dan-betmate** [comment]
I had this exact issue and tried

removeClippedSubviews={true}

and this worked for me.

I also tried the guide that @Isaccobosio linked above, but setting removeClippedSubviews to true was what I needed

**Comment by janetranada** [comment]
@Romick2005 thanks for the fix. My carousel in android kept jumping to one specific index on fast scroll. This seems to have handled the buggy scroll snap in android. However, i noticed a few things.

I'm using the default style of the carousel, with 7 cards on a loop:

1. Before, user scroll always moves 1 card at a time (which is what i need and want). With b4 beta version, it jumps 2 or 3 cards/items on fast scroll and 1 card/item on slow scroll. Are there some props I should set so that any scroll (fast or slow) will move the cards/items just one at a time?
2. Callback `onBeforeSnapToItem` is not called anymore. Callback `onSnapToItem` still works though. Was `onBeforeSnapToItem` affected by the change? 

I already tried putting the following props:
```
  enableMomentum={false}
  decelerationRate={'fast'}
```

**Comment by Romick2005** [comment]
@janetranada 
1) I cannot spot any properties for slow/fast scroll in the source, but I think you can modify [_getActiveItem](https://github.com/meliorence/react-native-snap-carousel/blob/59fa8950eeb8446124454ab3e2711038066115bb/src/carousel/Carousel.tsx#L702) method that is use to calculate nextActiveItem based on user action scroll distance.
2) Yep those method where [removed](https://github.com/meliorence/react-native-snap-carousel/blob/59fa8950eeb8446124454ab3e2711038066115bb/src/carousel/Carousel.tsx#L343). But I am using force back to the required next active slide index like `this.carouselRef.current.snapToItem(index)`. For my particular case, it is ok to slide from first slide to second and then do validation. If validation not passed I do programatically slide back using snapToItem.

**Comment by Bruce-Ming** [comment]
yes, react native 0.63.4 is ok,but 0.66.3 is broken it is react native version problem

**Comment by richard-grilla** [comment]
It's still not functioning properly on Android with 0.66.3

**Comment by richard-grilla** [comment]
Not sure if this helps anyone, but I did an edit to the `Carousel.js`, line `567` which made it at least useable.  I removed the `centerOffset`.

`        if (this._positions[lastIndex] && center> this._positions[lastIndex].end) {
`

**Comment by jobberma** [comment]
> @Romick2005 thanks for the fix. My carousel in android kept jumping to one specific index on fast scroll. This seems to have handled the buggy scroll snap in android. However, i noticed a few things.
> 
> I'm using the default style of the carousel, with 7 cards on a loop:
> 
> 1. Before, user scroll always moves 1 card at a time (which is what i need and want). With b4 beta version, it jumps 2 or 3 cards/items on fast scroll and 1 card/item on slow scroll. Are there some props I should set so that any scroll (fast or slow) will move the cards/items just one at a time?
> 2. Callback `onBeforeSnapToItem` is not called anymore. Callback `onSnapToItem` still works though. Was `onBeforeSnapToItem` affected by the change?
> 
> I already tried putting the following props:
> 
> ```
>   enableMomentum={false}
>   decelerationRate={'fast'}
> ```

I think it's late but it may help other people. For beta6, you can use ```disableIntervalMomentum``` to jump item by item even if you swipe fast.

**Comment by syedtehrimabbas** [comment]
> Hi, as it said in [their guide](https://github.com/meliorence/react-native-snap-carousel/blob/master/doc/TIPS_AND_TRICKS.md#momentum), you should put those props
> 
> ```
> ...
> enableMomentum={true}
> decelerationRate={0.9}
> ...
> ```
> 
> Hope it helps. This is how I fix this problem

This works for me.
Thank you

**Comment by SMKH-PRO** [comment]
facing same issue

**Comment by johnny-washswat** [comment]
I had the same issue I solved by adding `pagingEnabled` and it worked well on android and ios

**Comment by Iamivan1996** [comment]
All of the solutions are not work for me.. if i scroll it quickyly, android will getting stuck and autoPlay will not work until i scroll it back manually 

"react-native": "0.67.3",
"react-native-snap-carousel": "3.9.1",

**Comment by dsound-zz** [comment]
Having the same issue:

"react-native":  "0.67.5",
"react-native-snap-carousel": "^3.9.1",

**Comment by Bruce-Ming** [comment]
It is recommended to use the[ react-native-reanimated-carousel ](https://github.com/dohooo/react-native-reanimated-carousel)library. The current library seems to be no longer maintained.

**Comment by Abdullah-Sheikh** [comment]
> I had the same issue I solved by adding `pagingEnabled` and it worked well on android and ios

This works for me 🚀, Thanks @johnny-washswat

**Comment by tGulevich** [comment]
> I had the same issue I solved by adding `pagingEnabled` and it worked well on android and ios

@johnny-washswat Thanks! 🔥 That's tottally what I need!

**Comment by Hamadahmad000** [comment]
I'm replicating it if i use snap carousel within `ScrollView` otherwise it’s working fine
